### ISSUE TITLE
Do not allow targeted new expression when type is not apparant

### ISSUE THREAD

**Issue by manfred-brands** [dotnet/roslyn#60309]
**Brief description:**

Targeted new expressions are great in declarations so we don't have to repeat the type twice.
However, it seems to be also allowed elsewhere where the type is not apparent.

```csharp
_field = new();
```

Describe your **code style** rule here.

This rule would be complementary to IDE0090 and could use the same setting (`csharp_style_implicit_object_creation_when_type_is_apparent`). It would suggest **Specify explicit type in new(...)**.

**Languages applicable:**

C# 9.0 and later.

**Code example that the analyzer should report:**

Both statements in the constructor would be reported:

```csharp
public class A
{
    private B _field;

    public A()
    {
        _field = new();
       Method(new());
    }

    public Method(C c)
    {
    }
}
```

**Additional information:**

Although `type_is_apparent` is a matter of contention, not seeing it at all in the expression would qualify.

**Documentation requirements:**

When this analyzer is implemented, it must be documented by following the steps at [Documentation for IDE CodeStyle analyzers](https://github.com/dotnet/roslyn/blob/main/docs/contributing/Documentation%20for%20IDE%20CodeStyle%20analyzers.md).

**Comment by jinujoseph** [comment]
In design meeting to understand the value add and likelihood on agreement on single interpretation of the apparant.

**Comment by manfred-brands** [comment]
_apparent_ should be taken from the definition already accepted and used for IDE0090.
If IDE0090 doesn't suggest to use targeted new, it is not apparent.
For this rule it would be the reverse, if targeted new is used outside the cases defined in IDE0090, it should fire.

**Comment by manfred-brands** [comment]
If accepted, I'm happy to implement the analyzer in this repository,

**Comment by vatsalyaagrawal** [comment]
Design meeting notes:
1. We have not heard much feedback from customers. Would like to hear more feedback on this. 
2. We would accept community contribution for this. 
3. We would need a detailed proposal (either from community or from the team) before this is ready for implementation.

**Comment by CyrusNajmabadi** [comment]
Closing out due to lack of feedback.
